### PR TITLE
Integrating rest API into user info recovery with notification in user portal

### PR DESCRIPTION
**Pull request by IndunilRathnayake**

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementEndpointConstants.java [hunk 1]
```diff
@@ -59,6 +59,14 @@
         public static final String CHALLENGE_QUESTION_2_CLAIM = "http://wso2.org/claims/challengeQuestion2";
     }
 
+    public static final class UserInfoRecovery {
+        public static final String REST_URL = "https://localhost:9443/account-recovery/";
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/IdentityManagementEndpointConstants.java]: Won't hardcoding the URL be a problem ?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/UserInfoRecoveryWithNotificationClient.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import org.wso2.carbon.identity.mgt.endpoint.serviceclient.beans.User;
+import org.wso2.carbon.identity.mgt.endpoint.serviceclient.client.proxy.api.UserInfoRecoveryNotification;
+
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class UserInfoRecoveryWithNotificationClient {
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/UserInfoRecoveryWithNotificationClient.java]: Add a class level comment. Check other classes as well.

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ResetPasswordRequest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.mgt.endpoint.serviceclient.beans;
+
+import javax.xml.bind.annotation.*;
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/serviceclient/beans/ResetPasswordRequest.java]: Avoid * imports. Check in other places as well.

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-notify.jsp [hunk 4]
```diff
@@ -18,30 +18,34 @@
...
-                                 IdentityManagementEndpointUtil.getPrintableError("Failed to send email notification.",
-                                                                                  "Cannot verify the user with given username or captcha answer.",
-                                                                                  verificationBean));
-            request.getRequestDispatcher("error.jsp").forward(request, response);
-            return;
-        }
+    String userStoreDomain = Utils.getUserStoreDomainName(username);
+    String tenantDomain = MultitenantUtils.getTenantDomain(username);
+
+    Response sendNotificationResponse = userInfoRecoveryWithNotificationClient.sendRecoveryNotification(username, tenantDomain, userStoreDomain);
```
Comment (omindu) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-notify.jsp]: Check column width.

**Comment by IndunilRathnayake** [comment]
Moved to https://github.com/wso2/carbon-identity-framework/pull/274

**Pull request closed without merging**